airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlCdcConnectorMetadataInjector.java

```diff
@@ -19,20 +19,22 @@
 
 public class MySqlCdcConnectorMetadataInjector implements CdcMetadataInjector<MysqlDebeziumStateAttributes> {
 
-  private final long emittedAtConverted;
+  private final Instant emittedAt;
+  final MysqlDebeziumStateAttributes debeziumStateAttributes;
 
   // This now makes this class stateful. Please make sure to use the same instance within a sync
   private final AtomicLong recordCounter = new AtomicLong(1);
   private static final long ONE_HUNDRED_MILLION = 100_000_000;
   private static MySqlCdcConnectorMetadataInjector mySqlCdcConnectorMetadataInjector;
 
-  private MySqlCdcConnectorMetadataInjector(final Instant emittedAt) {
-    this.emittedAtConverted = emittedAt.getEpochSecond() * ONE_HUNDRED_MILLION;
+  private MySqlCdcConnectorMetadataInjector(final Instant emittedAt, final MysqlDebeziumStateAttributes stateAttributes) {
+    this.emittedAt = emittedAt;
+    this.debeziumStateAttributes = stateAttributes;
   }
 
-  public static MySqlCdcConnectorMetadataInjector getInstance(final Instant emittedAt) {
+  public static MySqlCdcConnectorMetadataInjector getInstance(final Instant emittedAt, final MysqlDebeziumStateAttributes stateAttributes) {
     if (mySqlCdcConnectorMetadataInjector == null) {
```

Comment:
Getting an instance with a different emittedAt or another set of attributes would still get an existing instance.
maybe do some comparison? (I know this was there before)